File: 401-binary-watch/401-binary-watch.py
class Solution:
    def readBinaryWatch(self, turnedOn: int) -> List[str]:
        ans = []
        for i in range(12):
            for j in range(60):
                if bin(i).count('1') + bin(j).count('1') == turnedOn:
                    ans.append( str(i) + ':' + ('0' if j < 10 else '') + str(j) )
        return ans

File: 401-binary-watch/401-binary-watch.java
class Solution {
    public List<String> readBinaryWatch(int turnedOn) {
        List<String> ans = new LinkedList<>();
        for(int i = 0; i < 12; i++)
            for(int j = 0; j < 60; j++)
                if(Integer.bitCount(i) + Integer.bitCount(j) == turnedOn)
                    ans.add(Integer.toString(i) + ":" + (j >= 10 ? Integer.toString(j) : ("0" + Integer.toString(j))));
        return ans;
    }
}